sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -134,24 +131,31 @@
   /**
    * Returns a {@code WindowedValue} with the given value in no windows, and the default timestamp
    * and pane.
+   *
+   * @deprecated a value in no windows technical is not "in" a PCollection. It is allowed to drop it
+   * at any point, and there are various points in various runners where this will happen. If a
+   * hack is needed where no window can be provided, create a dummy window that will fail fast if it
+   * is used outside its intended context.
    */
+  @Deprecated
   public static <T> WindowedValue<T> valueInEmptyWindows(T value) {
     return new ValueInEmptyWindows<T>(value, PaneInfo.NO_FIRING);
   }
 
   /**
    * Returns a {@code WindowedValue} with the given value in no windows, and the default timestamp
    * and the specified pane.
+   *
+   * @deprecated a value in no windows technical is not "in" a PCollection. It is allowed to drop it
+   * at any point, and there are various points in various runners where this will happen. If a
+   * hack is needed where no window can be provided, create a dummy window that will fail fast if it
+   * is used outside its intended context.
```

Comment:
what does "used outside its intended context" mean? If you're trying to be precise, just state what is allowed.